app/models/service.rb

```diff
@@ -156,7 +156,11 @@ def request_type
   end
 
   def retireable?
-    parent.present? ? true : type.present?
+    if parent.present? && provisioned?
```

Comment:
Does the `provisioned?` check effect all services?  Meaning, if it is not provisioned can we immediately say it cannot be retired?

```ruby
def retireable?
  false unless provisioned?

  parent.present? ? true : type.present?
end
```

Otherwise, it seems odd that we jump to using `type.present?` if `parent.present?` is `true` but `provisioned?` is `false`.

I guess the other form of this method I might expect is:
```ruby
  def retireable?
    if parent.present?
      provisioned?
    else
      type.present?
    end
  end
```

Reply:
> Does the provisioned? check effect all services?

It should! https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/lifecycle_mixin.rb#L18

Reply:
Right.  So does that mean that you can immediately return `false` if the service is not provisioned?  My first example.

Reply:
Ugh, okay. So I can have something that's technically provisioned but the lifecycle column is false, so while it can technically be (validly) retired, the retireable check will fail and you won't be able to retire it.

Reply:
If a service is in the middle of retirement and failed, is it still considered provisioned? If someone went in and manually retired a vm from a service, is the service still considered provisioned? Could you conceivably then have a service with all parts and pieces gone that is still "provisioned"?

Reply:
I would suggest that we focus on the normal use-case (provision service, then retire service) first and look into what I would consider edge conditions as a follow up.  Seems like it would overcomplicate this PR by trying to handle everything here.

But to add thoughts on the questions above, I would think we could write a method to validate if all the parts of a service are retired.  Then this check would need to be run at the end of the retirement of a resource and the service could be processed at that time.  Or it could be done as a schedule event, but without digging to deep into it I like the first approach better.  I am sure there are more nuances here that would need to be flushed out.  Again, this would be a follow up effort.

Reply:
Thanks for creating the data migration @d-m-u. I'm sure we'll encounter other edge conditions, but I can't think of anything else right now. @lfu Do you have any concerns?

The initial use case for the lifecycle state is to prevent retirement from being called/run while the service is being provisioned. I believe the code changes here satisfy that requirement.